target/linux/mediatek/image/gen_banana_pi_img.sh

```diff
@@ -0,0 +1,148 @@
+# partitions.
+#
+# The second sector must contain the BRLYT header, and the
+# special preloader image goes in sector 4; 2KiB into the image.
+#
+# The preloader loads U-Boot from  sector 640; 320KiB into the image.
+# The location and the size (512KiB) are fixed and not read from
+# the partition table. We set up a partition for it merely for
+# our own convenience for upgrades, etc.
+#
+# The second partition is a FAT file system containing the kernel
+# image and a uEnv.txt file, which is provided to this script as
+# $4 (bootfs image). Its size is configurable with the
+# CONFIG_BANANA_PI_BOOT_PARTSIZE option; by default 32MiB.
+#
+# The root filesystem comes next in the third partition.
+#
+#
+#   ------------------------   Sector   Offset
+#   |  MBR + SDMMC_BOOT    |     0       0x0
+#   |----------------------|
+#   |     BRLYT header     |     1       0x200
+#   |----------------------|
+#   .                      .
+#   .                      .
+#   |----------------------|
+#   |                      |     4       0x800
+#   |                      |
+#   |     Preloader        |
+#   .                      .
+#   .                      .
+#   |                      |     639
+#   |----------------------|
+#   |   MBR partition #1   |     640     0x50000
+#   |                      |
+#   |       U-Boot         |
+#   .                      .
+#   .                      .
+#   |                      |     1663
+#   |----------------------|
+#   |   MBR partition #2   |
+#   |                      |
+#   |    FAT partition     |   ( BANANA_PI_BOOT_PARTSIZE
+#   .                      .     default 32MiB )
+#   .    (kernel, uEnv)    .
+#   |                      |
+#   |----------------------|
+#   |   MBR partition #3   |
+#   |                      |
+#   |   Root partition     |
+#   |                      |   ( TARGET_ROOTFS_PARTSIZE
+#   |  (squashfs+overlay   |     default 104MiB )
+#   .    or ext4, etc.)    .
+#   .                      .
+#   |                      |
+#   ------------------------
+#
+# For eMMC boot, everything up to and including the preloader must be
+# written to /dev/mmcblk0boot0, with the SDMMC_BOOT header changed to
+# read EMMC_BOOT\0 instead.
+#
+# The contents of the main eMMC are identical to the SD card layout,
+# with the preloader loading 512KiB of U-Boot starting at 0x50000.
+
+function usage() {
+    echo "SYNTAX: $0 sd <file> <preloader image> <u-boot image> <bootfs image> <rootfs image> <bootfs size> <rootfs size>"
+    echo " OR:    $0 emmc <file> <preloader image>"
+    exit 1
+}
+
+set -e
+
+PRELOADER_OFFSET=2     # 2KiB
+
+SDMMC_BOOT="SDMMC_BOOT\x00\x00\x01\x00\x00\x00\x00\x02\x00\x00"
+EMMC_BOOT="EMMC_BOOT\x00\x00\x00\x01\x00\x00\x00\x00\x02\x00\x00"
+BRLYT="\
+BRLYT\x00\x00\x00\x01\x00\x00\x00\x00\x08\x00\x00\
+\x00\x08\x00\x00\x42\x42\x42\x42\x08\x00\x01\x00\x00\x08\x00\x00\
+\x00\x08\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+
+case $1 in
+    sd)
```

Comment:
It's a matter of personal preference but I'd rather remove the indent as most [02_network](https://github.com/openwrt/openwrt/blob/master/target/linux/mpc85xx/base-files/etc/board.d/02_network#L13) do. And I would be happy if you unify all indents in this file into space or tab...